aws/resource_aws_redshift_snapshot_schedule_test.go

```diff
@@ -0,0 +1,455 @@
+				hasPrefix = true
+				break
+			}
+		}
+
+		if !hasPrefix {
+			log.Printf("[INFO] Skipping Delete Redshift Snapshot Schedule: %s", identifier)
+			continue
+		}
+
+		_, err := conn.DeleteSnapshotSchedule(&redshift.DeleteSnapshotScheduleInput{
+			ScheduleIdentifier: snapshotSchedules.ScheduleIdentifier,
+		})
+		if isAWSErr(err, redshift.ErrCodeSnapshotScheduleNotFoundFault, "") {
+			return nil
+		}
+		if err != nil {
+			return fmt.Errorf("Error deleting Redshift Snapshot Schedule %s: %s", identifier, err)
+		}
+	}
+
+	return nil
+}
+
+func TestAccAWSRedshiftSnapshotSchedule_basic(t *testing.T) {
+	var v redshift.SnapshotSchedule
+
+	rName := acctest.RandString(8)
+	resourceName := "aws_redshift_snapshot_schedule.default"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSRedshiftSnapshotScheduleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSRedshiftSnapshotScheduleConfig(rName, "rate(12 hours)"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRedshiftSnapshotScheduleExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "definitions.#", "1"),
+				),
+			},
+			{
+				Config: testAccAWSRedshiftSnapshotScheduleConfig(rName, "cron(30 12 *)"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRedshiftSnapshotScheduleExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "definitions.#", "1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"force_destroy",
+				},
+			},
+		},
+	})
+}
+
+func TestAccAWSRedshiftSnapshotSchedule_withMultipleDefinition(t *testing.T) {
+	var v redshift.SnapshotSchedule
+
+	rName := acctest.RandString(8)
+	resourceName := "aws_redshift_snapshot_schedule.default"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSRedshiftSnapshotScheduleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSRedshiftSnapshotScheduleConfigWithMultipleDefinition(rName, "cron(30 12 *)", "cron(15 6 *)"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRedshiftSnapshotScheduleExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "definitions.#", "2"),
+				),
+			},
+			{
+				Config: testAccAWSRedshiftSnapshotScheduleConfigWithMultipleDefinition(rName, "cron(30 8 *)", "cron(15 10 *)"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRedshiftSnapshotScheduleExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "definitions.#", "2"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"force_destroy",
```

Comment:
We can call `d.Set("force_destroy", false)` during the import function to prevent needing to ignoring it here. 👍